
<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta charset="utf-8" />
    <title>pipeline_tools package &#8212; SeMPI 2.0 documentation</title>
    <link rel="stylesheet" href="../_static/nature.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script type="text/javascript" src="../_static/jquery.js"></script>
    <script type="text/javascript" src="../_static/underscore.js"></script>
    <script type="text/javascript" src="../_static/doctools.js"></script>
    <script type="text/javascript" src="../_static/language_data.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" /> 
  </head><body>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="../py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li class="nav-item nav-item-0"><a href="../index.html">SeMPI 2.0 documentation</a> &#187;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="pipeline-tools-package">
<h1>pipeline_tools package<a class="headerlink" href="#pipeline-tools-package" title="Permalink to this headline">¶</a></h1>
<div class="section" id="subpackages">
<h2>Subpackages<a class="headerlink" href="#subpackages" title="Permalink to this headline">¶</a></h2>
<div class="toctree-wrapper compound">
<ul>
<li class="toctree-l1"><a class="reference internal" href="pipeline_tools.rdkit_utils.html">pipeline_tools.rdkit_utils package</a><ul>
<li class="toctree-l2"><a class="reference internal" href="pipeline_tools.rdkit_utils.html#submodules">Submodules</a></li>
<li class="toctree-l2"><a class="reference internal" href="pipeline_tools.rdkit_utils.html#module-pipeline_tools.rdkit_utils.DB_scoring">pipeline_tools.rdkit_utils.DB_scoring module</a></li>
<li class="toctree-l2"><a class="reference internal" href="pipeline_tools.rdkit_utils.html#module-pipeline_tools.rdkit_utils.DB_set_up">pipeline_tools.rdkit_utils.DB_set_up module</a></li>
<li class="toctree-l2"><a class="reference internal" href="pipeline_tools.rdkit_utils.html#module-pipeline_tools.rdkit_utils.color_tools">pipeline_tools.rdkit_utils.color_tools module</a></li>
<li class="toctree-l2"><a class="reference internal" href="pipeline_tools.rdkit_utils.html#module-pipeline_tools.rdkit_utils.drawing_tools">pipeline_tools.rdkit_utils.drawing_tools module</a></li>
<li class="toctree-l2"><a class="reference internal" href="pipeline_tools.rdkit_utils.html#module-pipeline_tools.rdkit_utils">Module contents</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="pipeline_tools.specificity.html">pipeline_tools.specificity package</a><ul>
<li class="toctree-l2"><a class="reference internal" href="pipeline_tools.specificity.html#submodules">Submodules</a></li>
<li class="toctree-l2"><a class="reference internal" href="pipeline_tools.specificity.html#module-pipeline_tools.specificity.ml_utils">pipeline_tools.specificity.ml_utils module</a></li>
<li class="toctree-l2"><a class="reference internal" href="pipeline_tools.specificity.html#module-pipeline_tools.specificity.prediction">pipeline_tools.specificity.prediction module</a></li>
<li class="toctree-l2"><a class="reference internal" href="pipeline_tools.specificity.html#pipeline-tools-specificity-prediction-creator-module">pipeline_tools.specificity.prediction_creator module</a></li>
<li class="toctree-l2"><a class="reference internal" href="pipeline_tools.specificity.html#module-pipeline_tools.specificity.prediction_handler">pipeline_tools.specificity.prediction_handler module</a></li>
<li class="toctree-l2"><a class="reference internal" href="pipeline_tools.specificity.html#module-pipeline_tools.specificity.seq_utils">pipeline_tools.specificity.seq_utils module</a></li>
<li class="toctree-l2"><a class="reference internal" href="pipeline_tools.specificity.html#module-pipeline_tools.specificity">Module contents</a></li>
</ul>
</li>
</ul>
</div>
</div>
<div class="section" id="submodules">
<h2>Submodules<a class="headerlink" href="#submodules" title="Permalink to this headline">¶</a></h2>
</div>
<div class="section" id="module-pipeline_tools.cluster_prediction">
<span id="pipeline-tools-cluster-prediction-module"></span><h2>pipeline_tools.cluster_prediction module<a class="headerlink" href="#module-pipeline_tools.cluster_prediction" title="Permalink to this headline">¶</a></h2>
<dl class="function">
<dt id="pipeline_tools.cluster_prediction.combine_blocks">
<code class="descclassname">pipeline_tools.cluster_prediction.</code><code class="descname">combine_blocks</code><span class="sig-paren">(</span><em>df</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#pipeline_tools.cluster_prediction.combine_blocks" title="Permalink to this definition">¶</a></dt>
<dd><p>Combines blocks to clusters based on the cluster_dist kwarg</p>
<dl class="simple">
<dt>df<span class="classifier">pandas df</span></dt><dd><p>Input df</p>
</dd>
</dl>
<dl class="simple">
<dt>df<span class="classifier">pandas df</span></dt><dd><p>dataframe with cluster_id</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="pipeline_tools.cluster_prediction.get_functional_cluster">
<code class="descclassname">pipeline_tools.cluster_prediction.</code><code class="descname">get_functional_cluster</code><span class="sig-paren">(</span><em>df</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#pipeline_tools.cluster_prediction.get_functional_cluster" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates clusters for the final output, only functional clusters are take,
currently the definition of a functional cluster is, that it contains at 
least one functional module, a more specific definition might also be envisaged</p>
<dl class="simple">
<dt>df<span class="classifier">pandas df</span></dt><dd><p>Input df</p>
</dd>
</dl>
<dl class="simple">
<dt>cluster_df<span class="classifier">pandas df</span></dt><dd><p>Dataframe with nice clusters, shortened for the final output</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="pipeline_tools.cluster_prediction.get_functional_cluster_smiles_scoring">
<code class="descclassname">pipeline_tools.cluster_prediction.</code><code class="descname">get_functional_cluster_smiles_scoring</code><span class="sig-paren">(</span><em>cluster_df</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#pipeline_tools.cluster_prediction.get_functional_cluster_smiles_scoring" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</div>
<div class="section" id="module-pipeline_tools.database_handling">
<span id="pipeline-tools-database-handling-module"></span><h2>pipeline_tools.database_handling module<a class="headerlink" href="#module-pipeline_tools.database_handling" title="Permalink to this headline">¶</a></h2>
<dl class="function">
<dt id="pipeline_tools.database_handling.df2mysql">
<code class="descclassname">pipeline_tools.database_handling.</code><code class="descname">df2mysql</code><span class="sig-paren">(</span><em>df</em>, <em>file_name='gbk_df.db'</em>, <em>sql_id=None</em>, <em>record_id=None</em>, <em>record_name=None</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#pipeline_tools.database_handling.df2mysql" title="Permalink to this definition">¶</a></dt>
<dd><p>Converts a df to a mysql database</p>
<dl class="simple">
<dt>df<span class="classifier">pandas df</span></dt><dd><p>dataframe of the gbk</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="pipeline_tools.database_handling.df2postgres">
<code class="descclassname">pipeline_tools.database_handling.</code><code class="descname">df2postgres</code><span class="sig-paren">(</span><em>df</em>, <em>table_name=''</em>, <em>sql_id=None</em>, <em>record_id=None</em>, <em>record_name=None</em>, <em>engine_string='postgresql://rdkit_db_user:rdkit&#64;localhost:5432/rdkit_db'</em>, <em>sql_overwrite=False</em>, <em>db_input=None</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#pipeline_tools.database_handling.df2postgres" title="Permalink to this definition">¶</a></dt>
<dd><p>Stores the DB in postgres DB</p>
<dl class="simple">
<dt>df<span class="classifier">pandas df</span></dt><dd><p>dataframe of the gbk</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="pipeline_tools.database_handling.dummy_input_file">
<code class="descclassname">pipeline_tools.database_handling.</code><code class="descname">dummy_input_file</code><span class="sig-paren">(</span><em>table_name='sempi_input_file'</em>, <em>database='rdkit_db'</em>, <em>user='rdkit_db_user'</em>, <em>password='rdkit'</em>, <em>host='localhost'</em>, <em>db_storage=''</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#pipeline_tools.database_handling.dummy_input_file" title="Permalink to this definition">¶</a></dt>
<dd><p>When the pipeline is called for the server 
the sempi_input_file relation is created by django,
for the non-server run, an dummy relation is created,
so that foreign keys work.</p>
</dd></dl>

<dl class="function">
<dt id="pipeline_tools.database_handling.project2zip">
<code class="descclassname">pipeline_tools.database_handling.</code><code class="descname">project2zip</code><span class="sig-paren">(</span><em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#pipeline_tools.database_handling.project2zip" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates a zip archive of the project in the project</p>
<dl class="simple">
<dt>project_path<span class="classifier">string</span></dt><dd><p>path to the project</p>
</dd>
</dl>
</dd></dl>

</div>
<div class="section" id="module-pipeline_tools.dataframe_curation">
<span id="pipeline-tools-dataframe-curation-module"></span><h2>pipeline_tools.dataframe_curation module<a class="headerlink" href="#module-pipeline_tools.dataframe_curation" title="Permalink to this headline">¶</a></h2>
<dl class="function">
<dt id="pipeline_tools.dataframe_curation.add_reorder_based_on_strand">
<code class="descclassname">pipeline_tools.dataframe_curation.</code><code class="descname">add_reorder_based_on_strand</code><span class="sig-paren">(</span><em>df</em>, <em>based_on='block_id'</em><span class="sig-paren">)</span><a class="headerlink" href="#pipeline_tools.dataframe_curation.add_reorder_based_on_strand" title="Permalink to this definition">¶</a></dt>
<dd><p>Add the corr_domai_id column to the df,
the new order is based on each locus tag,
ordered by the strand:</p>
<dl class="simple">
<dt>df<span class="classifier">pandas df</span></dt><dd><p>dataframe of the gbk</p>
</dd>
</dl>
<dl class="simple">
<dt>df<span class="classifier">pandas df</span></dt><dd><p>dataframe of the gbk (with new column)</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="pipeline_tools.dataframe_curation.combine_close_doms">
<code class="descclassname">pipeline_tools.dataframe_curation.</code><code class="descname">combine_close_doms</code><span class="sig-paren">(</span><em>df</em>, <em>hmm_path</em><span class="sig-paren">)</span><a class="headerlink" href="#pipeline_tools.dataframe_curation.combine_close_doms" title="Permalink to this definition">¶</a></dt>
<dd><p>Combine split up domains
seq A , seq B - &gt; remove(seq B) ,(combination of A + B)</p>
<dl class="simple">
<dt>df<span class="classifier">pandas df</span></dt><dd><p>Input df</p>
</dd>
<dt>hmm_path<span class="classifier">string</span></dt><dd><p>Path of the folder with the hmmer profiles (.hmm)</p>
</dd>
</dl>
<dl class="simple">
<dt>df<span class="classifier">pandas df</span></dt><dd><p>Input df with column ‘duplicate’, domains are joined and old domain removed</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="pipeline_tools.dataframe_curation.df_bool_trimming">
<code class="descclassname">pipeline_tools.dataframe_curation.</code><code class="descname">df_bool_trimming</code><span class="sig-paren">(</span><em>df</em><span class="sig-paren">)</span><a class="headerlink" href="#pipeline_tools.dataframe_curation.df_bool_trimming" title="Permalink to this definition">¶</a></dt>
<dd><p>Somehow some columns get converted from 1,0 to True or False, messes up the DB storage !</p>
</dd></dl>

<dl class="function">
<dt id="pipeline_tools.dataframe_curation.float2int">
<code class="descclassname">pipeline_tools.dataframe_curation.</code><code class="descname">float2int</code><span class="sig-paren">(</span><em>df_list</em><span class="sig-paren">)</span><a class="headerlink" href="#pipeline_tools.dataframe_curation.float2int" title="Permalink to this definition">¶</a></dt>
<dd><p>For the web front end, ints are better then floats, if possible (for example for ids)
If a column or the index in a df contains only values that could be int 
it is converted, if not it is kept</p>
</dd></dl>

<dl class="function">
<dt id="pipeline_tools.dataframe_curation.gbk2df">
<code class="descclassname">pipeline_tools.dataframe_curation.</code><code class="descname">gbk2df</code><span class="sig-paren">(</span><em>GBK_PATH</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#pipeline_tools.dataframe_curation.gbk2df" title="Permalink to this definition">¶</a></dt>
<dd><p>Converts the gbk file produced 
in the genome_upload 
part into a pandas dataframe, all the fields below need to 
exists.</p>
<dl class="simple">
<dt>GBK_PATH<span class="classifier">string</span></dt><dd><p>Path of the input gbk file</p>
</dd>
</dl>
<dl class="simple">
<dt>df<span class="classifier">pandas df</span></dt><dd><p>dataframe of the gbk</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="pipeline_tools.dataframe_curation.hmmer_length_dict">
<code class="descclassname">pipeline_tools.dataframe_curation.</code><code class="descname">hmmer_length_dict</code><span class="sig-paren">(</span><em>hmm_path</em><span class="sig-paren">)</span><a class="headerlink" href="#pipeline_tools.dataframe_curation.hmmer_length_dict" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates a dict of hmm profiles with their corresponding lengths, can be used to check if
domains are duplicates.</p>
<dl class="simple">
<dt>hmm_path<span class="classifier">string</span></dt><dd><p>Path of the folder with the hmmer profiles (.hmm)</p>
</dd>
</dl>
<dl class="simple">
<dt>hmm_lengths: dict</dt><dd><p>key -&gt; hmm profile, key -&gt; length</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="pipeline_tools.dataframe_curation.join_close_genes">
<code class="descclassname">pipeline_tools.dataframe_curation.</code><code class="descname">join_close_genes</code><span class="sig-paren">(</span><em>df</em><span class="sig-paren">)</span><a class="headerlink" href="#pipeline_tools.dataframe_curation.join_close_genes" title="Permalink to this definition">¶</a></dt>
<dd><p>Some genes are split, although they seem to be part of the 
same BGC assembly line.
For the detection of modules,
close genes, with BGC_domains and same strand are joined together into
locus_blocks with new id block_id.</p>
<dl class="simple">
<dt>df<span class="classifier">pandas df</span></dt><dd><p>Input df</p>
</dd>
</dl>
<dl class="simple">
<dt>df<span class="classifier">pandas df</span></dt><dd><p>Input df with column ‘block_id’</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="pipeline_tools.dataframe_curation.remove_overlap">
<code class="descclassname">pipeline_tools.dataframe_curation.</code><code class="descname">remove_overlap</code><span class="sig-paren">(</span><em>df</em><span class="sig-paren">)</span><a class="headerlink" href="#pipeline_tools.dataframe_curation.remove_overlap" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</div>
<div class="section" id="module-pipeline_tools.file_handling">
<span id="pipeline-tools-file-handling-module"></span><h2>pipeline_tools.file_handling module<a class="headerlink" href="#module-pipeline_tools.file_handling" title="Permalink to this headline">¶</a></h2>
<dl class="function">
<dt id="pipeline_tools.file_handling.create_pipeline_meta_df">
<code class="descclassname">pipeline_tools.file_handling.</code><code class="descname">create_pipeline_meta_df</code><span class="sig-paren">(</span><em>set_genome_browser</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#pipeline_tools.file_handling.create_pipeline_meta_df" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates meta infos and paths for the database, which is
shown for each record on the website view,
the genome browser is optional, should not be created when 
there are no proteins.</p>
</dd></dl>

<dl class="function">
<dt id="pipeline_tools.file_handling.gbk_sanity_check">
<code class="descclassname">pipeline_tools.file_handling.</code><code class="descname">gbk_sanity_check</code><span class="sig-paren">(</span><em>gbk_record</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#pipeline_tools.file_handling.gbk_sanity_check" title="Permalink to this definition">¶</a></dt>
<dd><p>Checks if the gbk file can be used as it is for the downstream pipeline
Detail: Checks if the CDS feature has all needed sub features</p>
<dl class="simple">
<dt>gbk_record<span class="classifier">gbk_record</span></dt><dd><p>Parsed gbk</p>
</dd>
</dl>
<dl class="simple">
<dt>gbk_path<span class="classifier">sting</span></dt><dd><p>If the record can be passed on the pipe the file_path is passed</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="pipeline_tools.file_handling.input_parsing_initial">
<code class="descclassname">pipeline_tools.file_handling.</code><code class="descname">input_parsing_initial</code><span class="sig-paren">(</span><em>input_file</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#pipeline_tools.file_handling.input_parsing_initial" title="Permalink to this definition">¶</a></dt>
<dd><p>Parses the input file, 
checks the file type and 
if the file can be parsed,
create a mapper which
allows to process each record as a single record 
downstream of the pipeline.</p>
<dl class="simple">
<dt>input_file<span class="classifier">file</span></dt><dd><p>Input file, either gbk or fasta</p>
</dd>
</dl>
<dl class="simple">
<dt>multi_record_file_mapping<span class="classifier">dict</span></dt><dd><p>key: index of the record, value tuple (new input path, new output path)</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="pipeline_tools.file_handling.input_parsing_single">
<code class="descclassname">pipeline_tools.file_handling.</code><code class="descname">input_parsing_single</code><span class="sig-paren">(</span><em>input_file</em>, <em>f_type</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#pipeline_tools.file_handling.input_parsing_single" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="function">
<dt id="pipeline_tools.file_handling.prepare_multi_entry_mode">
<code class="descclassname">pipeline_tools.file_handling.</code><code class="descname">prepare_multi_entry_mode</code><span class="sig-paren">(</span><em>file</em>, <em>f_type</em>, <em>tmp_folder</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#pipeline_tools.file_handling.prepare_multi_entry_mode" title="Permalink to this definition">¶</a></dt>
<dd><p>In order to run multi entry files, the file is split and
passed to the pipeline iteratively, each is processed in a subfolder in the tmp_folder</p>
</dd></dl>

</div>
<div class="section" id="module-pipeline_tools.gbk_output_generation">
<span id="pipeline-tools-gbk-output-generation-module"></span><h2>pipeline_tools.gbk_output_generation module<a class="headerlink" href="#module-pipeline_tools.gbk_output_generation" title="Permalink to this headline">¶</a></h2>
<dl class="function">
<dt id="pipeline_tools.gbk_output_generation.df2gbk">
<code class="descclassname">pipeline_tools.gbk_output_generation.</code><code class="descname">df2gbk</code><span class="sig-paren">(</span><em>df</em>, <em>TEMPLATE_GBK_PATH</em>, <em>sempi_credentials</em>, <em>only_domains=False</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#pipeline_tools.gbk_output_generation.df2gbk" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates gbk for the entire genome and the clusters</p>
<dl class="simple">
<dt>df<span class="classifier">pandas df</span></dt><dd><p>dataframe of domains</p>
</dd>
<dt>TEMPLATE_GBK_PATH<span class="classifier">string</span></dt><dd><p>templete genbank file used to create the output files
header stays the same</p>
</dd>
</dl>
<dl class="simple">
<dt>OUTPUT_GBK_PATH<span class="classifier">string</span></dt><dd><p>path to the final_genome.gbk</p>
</dd>
<dt>OUTPUT_FOLDER<span class="classifier">string</span></dt><dd><p>path to the folder with the clusters</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="pipeline_tools.gbk_output_generation.df2gbk_features">
<code class="descclassname">pipeline_tools.gbk_output_generation.</code><code class="descname">df2gbk_features</code><span class="sig-paren">(</span><em>df</em>, <em>start_column='loc_start'</em>, <em>end_column='loc_end'</em>, <em>strand_column='strand'</em>, <em>feature_type='BGC Domain'</em>, <em>exclude=[]</em><span class="sig-paren">)</span><a class="headerlink" href="#pipeline_tools.gbk_output_generation.df2gbk_features" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates a list of biopython features from a df</p>
<dl class="simple">
<dt>df<span class="classifier">pandas df</span></dt><dd><p>dataframe of the gbk</p>
</dd>
<dt>start_column<span class="classifier">string</span></dt><dd><p>column to use for the start of the feature (must be int)</p>
</dd>
<dt>end_column<span class="classifier">string</span></dt><dd><p>see start_column</p>
</dd>
<dt>strand_column<span class="classifier">string</span></dt><dd><p>see start_column</p>
</dd>
<dt>feature_type<span class="classifier">string</span></dt><dd><p>This will be the name of the feature in the GBK</p>
</dd>
<dt>exclude<span class="classifier">list</span></dt><dd><p>All additional columns are used as feature.qualifiers
if not excluded</p>
</dd>
</dl>
<dl class="simple">
<dt>feature_list<span class="classifier">list</span></dt><dd><p>List of created features</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="pipeline_tools.gbk_output_generation.get_blocks">
<code class="descclassname">pipeline_tools.gbk_output_generation.</code><code class="descname">get_blocks</code><span class="sig-paren">(</span><em>df</em><span class="sig-paren">)</span><a class="headerlink" href="#pipeline_tools.gbk_output_generation.get_blocks" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates a flattened df for the blocks (only infos needed for the blocks)
to use to create features for the blocks</p>
<dl class="simple">
<dt>df<span class="classifier">pandas df</span></dt><dd><p>dataframe of modules</p>
</dd>
</dl>
<dl class="simple">
<dt>feature_df<span class="classifier">pandas df</span></dt><dd><p>df for the blocks</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="pipeline_tools.gbk_output_generation.get_cluster">
<code class="descclassname">pipeline_tools.gbk_output_generation.</code><code class="descname">get_cluster</code><span class="sig-paren">(</span><em>df</em><span class="sig-paren">)</span><a class="headerlink" href="#pipeline_tools.gbk_output_generation.get_cluster" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates a flattened df for the clusters (only infos needed for the clusters)
to use to create features for the clusters</p>
<dl class="simple">
<dt>df<span class="classifier">pandas df</span></dt><dd><p>dataframe of blocks</p>
</dd>
</dl>
<dl class="simple">
<dt>feature_df<span class="classifier">pandas df</span></dt><dd><p>df for the clusters</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="pipeline_tools.gbk_output_generation.get_domains">
<code class="descclassname">pipeline_tools.gbk_output_generation.</code><code class="descname">get_domains</code><span class="sig-paren">(</span><em>df</em><span class="sig-paren">)</span><a class="headerlink" href="#pipeline_tools.gbk_output_generation.get_domains" title="Permalink to this definition">¶</a></dt>
<dd><p>The domains need locus tag as name to create nice genome browsers</p>
<dl class="simple">
<dt>df<span class="classifier">pandas df</span></dt><dd><p>dataframe of domains</p>
</dd>
</dl>
<dl class="simple">
<dt>new_df<span class="classifier">pandas df</span></dt><dd><p>df with changes column names</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="pipeline_tools.gbk_output_generation.get_modules">
<code class="descclassname">pipeline_tools.gbk_output_generation.</code><code class="descname">get_modules</code><span class="sig-paren">(</span><em>df</em><span class="sig-paren">)</span><a class="headerlink" href="#pipeline_tools.gbk_output_generation.get_modules" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates a flattened df for the modules (only infos needed for the modules)
to use to create features for the modules</p>
<dl class="simple">
<dt>df<span class="classifier">pandas df</span></dt><dd><p>dataframe of domains</p>
</dd>
</dl>
<dl class="simple">
<dt>feature_df<span class="classifier">pandas df</span></dt><dd><p>df for the modules</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="pipeline_tools.gbk_output_generation.split_features_per_cluster">
<code class="descclassname">pipeline_tools.gbk_output_generation.</code><code class="descname">split_features_per_cluster</code><span class="sig-paren">(</span><em>gbk</em>, <em>OUTPUT_GBK_PATH</em>, <em>sempi_credentials</em>, <em>overlap=10000</em><span class="sig-paren">)</span><a class="headerlink" href="#pipeline_tools.gbk_output_generation.split_features_per_cluster" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates a gbk file for each cluster, by splitting the features according to 
the position of the clusters (additional 1000 pb are added to the cluster to avoid include the sides)</p>
<dl class="simple">
<dt>gbk<span class="classifier">biopython genbank record</span></dt><dd><p>Used as template for the clusters gbks</p>
</dd>
<dt>OUTPUT_GBK_PATH<span class="classifier">string</span></dt><dd><p>The output folder is created in the same folder as the genome gbk</p>
</dd>
</dl>
<dl class="simple">
<dt>OUTPUT_FOLDER<span class="classifier">string</span></dt><dd><p>path to the cluster folder</p>
</dd>
</dl>
</dd></dl>

</div>
<div class="section" id="module-pipeline_tools.gene_prediction">
<span id="pipeline-tools-gene-prediction-module"></span><h2>pipeline_tools.gene_prediction module<a class="headerlink" href="#module-pipeline_tools.gene_prediction" title="Permalink to this headline">¶</a></h2>
<p>This module creates the gene prediction for the submitted gene, if it is not already done</p>
<dl class="function">
<dt id="pipeline_tools.gene_prediction.prodigal_parser">
<code class="descclassname">pipeline_tools.gene_prediction.</code><code class="descname">prodigal_parser</code><span class="sig-paren">(</span><em>sequence</em>, <em>seq_name</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#pipeline_tools.gene_prediction.prodigal_parser" title="Permalink to this definition">¶</a></dt>
<dd><p>calls the prodigal gene prediction tool as a subprocess, creates a genbank file</p>
</dd></dl>

</div>
<div class="section" id="module-pipeline_tools.genome_diagram">
<span id="pipeline-tools-genome-diagram-module"></span><h2>pipeline_tools.genome_diagram module<a class="headerlink" href="#module-pipeline_tools.genome_diagram" title="Permalink to this headline">¶</a></h2>
<dl class="function">
<dt id="pipeline_tools.genome_diagram.blockPrinting">
<code class="descclassname">pipeline_tools.genome_diagram.</code><code class="descname">blockPrinting</code><span class="sig-paren">(</span><em>func</em><span class="sig-paren">)</span><a class="headerlink" href="#pipeline_tools.genome_diagram.blockPrinting" title="Permalink to this definition">¶</a></dt>
<dd><p>decorater used to block function printing to the console
taken from <a class="reference external" href="https://stackoverflow.com/questions/8391411/suppress-calls-to-print-python">https://stackoverflow.com/questions/8391411/suppress-calls-to-print-python</a></p>
</dd></dl>

<dl class="function">
<dt id="pipeline_tools.genome_diagram.change_path_index_gbrowser">
<code class="descclassname">pipeline_tools.genome_diagram.</code><code class="descname">change_path_index_gbrowser</code><span class="sig-paren">(</span><em>file_path=None</em><span class="sig-paren">)</span><a class="headerlink" href="#pipeline_tools.genome_diagram.change_path_index_gbrowser" title="Permalink to this definition">¶</a></dt>
<dd><p>Changes the content of the index file, so that the 
scripts are correctly loaded</p>
<dl class="simple">
<dt>file_path<span class="classifier">string</span></dt><dd><p>Path to the index file</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="pipeline_tools.genome_diagram.draw_diagram">
<code class="descclassname">pipeline_tools.genome_diagram.</code><code class="descname">draw_diagram</code><span class="sig-paren">(</span><em>GBK_PATH, area=[0, 100], positions=None, cluster_name=None, fragments=10, **kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#pipeline_tools.genome_diagram.draw_diagram" title="Permalink to this definition">¶</a></dt>
<dd><p>draws a digram of the genome</p>
</dd></dl>

<dl class="function">
<dt id="pipeline_tools.genome_diagram.gbk2genomebrowser">
<code class="descclassname">pipeline_tools.genome_diagram.</code><code class="descname">gbk2genomebrowser</code><span class="sig-paren">(</span><em>GBK_PATH</em>, <em>OUTPUT_FOLDER</em><span class="sig-paren">)</span><a class="headerlink" href="#pipeline_tools.genome_diagram.gbk2genomebrowser" title="Permalink to this definition">¶</a></dt>
<dd><p>Wrapper for the D3GB.gbk2genomebrowser, 
creates browsers for the entire genome and the clusters
Creates only one scrips folder</p>
<dl class="simple">
<dt>GBK_PATH<span class="classifier">string</span></dt><dd><p>Path to the main gbk file</p>
</dd>
<dt>OUTPUT_FOLDER<span class="classifier">string</span></dt><dd><p>Path to the cluster gbk files</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="pipeline_tools.genome_diagram.gbk2genomebrowser_no_print">
<code class="descclassname">pipeline_tools.genome_diagram.</code><code class="descname">gbk2genomebrowser_no_print</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#pipeline_tools.genome_diagram.gbk2genomebrowser_no_print" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="function">
<dt id="pipeline_tools.genome_diagram.get_domain_color_dict">
<code class="descclassname">pipeline_tools.genome_diagram.</code><code class="descname">get_domain_color_dict</code><span class="sig-paren">(</span><em>record</em>, <em>feature_type='BGC_Domain'</em><span class="sig-paren">)</span><a class="headerlink" href="#pipeline_tools.genome_diagram.get_domain_color_dict" title="Permalink to this definition">¶</a></dt>
<dd><p>automatically assignes come colors to the domains in the gbk</p>
</dd></dl>

</div>
<div class="section" id="module-pipeline_tools.module_prediction">
<span id="pipeline-tools-module-prediction-module"></span><h2>pipeline_tools.module_prediction module<a class="headerlink" href="#module-pipeline_tools.module_prediction" title="Permalink to this headline">¶</a></h2>
<dl class="function">
<dt id="pipeline_tools.module_prediction.assign_end_part">
<code class="descclassname">pipeline_tools.module_prediction.</code><code class="descname">assign_end_part</code><span class="sig-paren">(</span><em>df</em><span class="sig-paren">)</span><a class="headerlink" href="#pipeline_tools.module_prediction.assign_end_part" title="Permalink to this definition">¶</a></dt>
<dd><p>Assigns a new reduction type to an end part of pks and nrps,
same type
this must be called after the reduction shift,
nice carboxyl group !</p>
<p>The condition is given by the TE domain.</p>
<dl class="simple">
<dt>df<span class="classifier">pandas df</span></dt><dd><p>gbk df</p>
</dd>
</dl>
<dl class="simple">
<dt>df<span class="classifier">pandas df</span></dt><dd><p>gbk df with changed ‘reduction’ column for the end part</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="pipeline_tools.module_prediction.block_module_index">
<code class="descclassname">pipeline_tools.module_prediction.</code><code class="descname">block_module_index</code><span class="sig-paren">(</span><em>df</em><span class="sig-paren">)</span><a class="headerlink" href="#pipeline_tools.module_prediction.block_module_index" title="Permalink to this definition">¶</a></dt>
<dd><p>The module indices are unique but unsorted as returned by get_module_types,
this function sorts the indices by appearance in a block</p>
<dl class="simple">
<dt>locus_df<span class="classifier">pandas df</span></dt><dd><p>dataframe with unsorted indices</p>
</dd>
</dl>
<dl class="simple">
<dt>locus_df<span class="classifier">pandas df</span></dt><dd><p>dataframe with sorted indices</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="pipeline_tools.module_prediction.nrps_detection">
<code class="descclassname">pipeline_tools.module_prediction.</code><code class="descname">nrps_detection</code><span class="sig-paren">(</span><em>df</em><span class="sig-paren">)</span><a class="headerlink" href="#pipeline_tools.module_prediction.nrps_detection" title="Permalink to this definition">¶</a></dt>
<dd><p>Detection of the nrps modules</p>
<dl class="simple">
<dt>df<span class="classifier">pandas df</span></dt><dd><p>dataframe of the gbk</p>
</dd>
</dl>
<p>df : pandas df</p>
<blockquote>
<div><p>dataframe of the gbk (with new columns):
Columns:    module: is a module ? (bool)</p>
<blockquote>
<div><p>module_type: which ? (pks, nrps, none)
red_type: 
module_index: order on the locus, based on corr domain id
end_part: (bool)
trans_AT: (bool)
MT: (bool)
M_start: (int)
M_end: (int)</p>
</div></blockquote>
</div></blockquote>
</dd></dl>

<dl class="function">
<dt id="pipeline_tools.module_prediction.pks_detection">
<code class="descclassname">pipeline_tools.module_prediction.</code><code class="descname">pks_detection</code><span class="sig-paren">(</span><em>df</em>, <em>allow_trans_AT=True</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#pipeline_tools.module_prediction.pks_detection" title="Permalink to this definition">¶</a></dt>
<dd><p>Detection of the pks modules</p>
<dl class="simple">
<dt>df<span class="classifier">pandas df</span></dt><dd><p>dataframe of the gbk</p>
</dd>
</dl>
<p>df : pandas df</p>
<blockquote>
<div><p>dataframe of the gbk (with new columns):
Columns:    module: is a module ? (bool)</p>
<blockquote>
<div><p>module_type: which ? (pks, nrps, none)
red_type: 
module_index: order on the locus, based on corr domain id
end_part: (bool)
trans_AT: (bool)
MT: (bool)
M_start: (int)
M_end: (int)</p>
</div></blockquote>
</div></blockquote>
</dd></dl>

<dl class="function">
<dt id="pipeline_tools.module_prediction.pks_reduction_shift">
<code class="descclassname">pipeline_tools.module_prediction.</code><code class="descname">pks_reduction_shift</code><span class="sig-paren">(</span><em>df</em><span class="sig-paren">)</span><a class="headerlink" href="#pipeline_tools.module_prediction.pks_reduction_shift" title="Permalink to this definition">¶</a></dt>
<dd><p>Note: Reduction shift is applied to nrps and pks, as the pks can propagate
the reduction to the nrps</p>
<p>The reduction pattern of one module is always applied
to the substrate of the previous module, therefore the reduction defined in the
modules needs to be shifted by one module</p>
<blockquote>
<div><dl>
<dt>load        hydroxly </dt><dd><div class="line-block">
<div class="line">|     </div>
</div>
</dd>
</dl>
</div></blockquote>
<p>KS-AT-ACP-KS-AT-KR-ACP</p>
<p>shift</p>
<blockquote>
<div><dl>
<dt>hydroxly       undefined</dt><dd><div class="line-block">
<div class="line">|      </div>
</div>
</dd>
</dl>
</div></blockquote>
<p>KS-AT-ACP-KS-AT-KR-ACP</p>
<dl class="simple">
<dt>df<span class="classifier">pandas df</span></dt><dd><p>gbk df</p>
</dd>
</dl>
<dl class="simple">
<dt>df<span class="classifier">pandas df</span></dt><dd><p>gbk df with changed ‘reduction’ column</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="pipeline_tools.module_prediction.regex_pattern_in_list">
<code class="descclassname">pipeline_tools.module_prediction.</code><code class="descname">regex_pattern_in_list</code><span class="sig-paren">(</span><em>liste</em>, <em>regex_pattern</em>, <em>add_index=None</em><span class="sig-paren">)</span><a class="headerlink" href="#pipeline_tools.module_prediction.regex_pattern_in_list" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates a regex from a list of elements, performs a 
pattern search and returns the indices of the items in
the list that matches the pattern, works as long as 
the items do match regex synthax, for example, x and xx as
items could lead to wrong match, therefore no duplicated 
or non unique items</p>
<dl class="simple">
<dt>liste<span class="classifier">list</span></dt><dd><p>list of string elements</p>
</dd>
<dt>regex_pattern<span class="classifier">string</span></dt><dd><p>regex to search in the list</p>
</dd>
<dt>add_index<span class="classifier">None / int</span></dt><dd><p>index can be adjusted to an offset</p>
</dd>
</dl>
<dl class="simple">
<dt>result_index_list<span class="classifier">list of lists of ints</span></dt><dd><p>list of list with matches of the regex in the liste</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="pipeline_tools.module_prediction.split_block_by_TE">
<code class="descclassname">pipeline_tools.module_prediction.</code><code class="descname">split_block_by_TE</code><span class="sig-paren">(</span><em>df</em><span class="sig-paren">)</span><a class="headerlink" href="#pipeline_tools.module_prediction.split_block_by_TE" title="Permalink to this definition">¶</a></dt>
<dd><p>Splits the block if there is a TE part in between. 
Continues blocks with TE do not make sense.
Happens quiet often for pks domains.</p>
<dl class="simple">
<dt>locus_df<span class="classifier">pandas df</span></dt><dd><p>dataframe with continues blocks</p>
</dd>
</dl>
<dl class="simple">
<dt>locus_df<span class="classifier">pandas df</span></dt><dd><p>dataframe with split blocks</p>
</dd>
</dl>
</dd></dl>

</div>
<div class="section" id="module-pipeline_tools.protein_prediction">
<span id="pipeline-tools-protein-prediction-module"></span><h2>pipeline_tools.protein_prediction module<a class="headerlink" href="#module-pipeline_tools.protein_prediction" title="Permalink to this headline">¶</a></h2>
<p>This module gets the hmmer3 searcher for the desired modules</p>
<dl class="function">
<dt id="pipeline_tools.protein_prediction.create_protein_fasta">
<code class="descclassname">pipeline_tools.protein_prediction.</code><code class="descname">create_protein_fasta</code><span class="sig-paren">(</span><em>gbk_path</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#pipeline_tools.protein_prediction.create_protein_fasta" title="Permalink to this definition">¶</a></dt>
<dd><p>Create protein fasta for HMMER search</p>
<dl class="simple">
<dt>gbk_path<span class="classifier">string</span></dt><dd><p>Path of the input gbk file</p>
</dd>
</dl>
<dl class="simple">
<dt>fasta_path<span class="classifier">string</span></dt><dd><p>Path of the created fasta file</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="pipeline_tools.protein_prediction.hmmer_wrapper">
<code class="descclassname">pipeline_tools.protein_prediction.</code><code class="descname">hmmer_wrapper</code><span class="sig-paren">(</span><em>FASTA_PATH</em>, <em>HMMER_PROFILES_PATH</em>, <em>HMMER_TRESHOLD_FILE</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#pipeline_tools.protein_prediction.hmmer_wrapper" title="Permalink to this definition">¶</a></dt>
<dd><p>HMMER search wrapper for multiple fasta</p>
<dl class="simple">
<dt>FASTA_PATH<span class="classifier">string</span></dt><dd><p>Multiple fasta file created with create_protein_fasta</p>
</dd>
<dt>HMMER_PROFILES_PATH<span class="classifier">string</span></dt><dd><p>Path of the profiles to be searched in the genome</p>
</dd>
<dt>HMMER_TRESHOLD_FILE<span class="classifier">string</span></dt><dd><p>Threhold file
Ex.: 
HMM_file,threshold
ACP.hmm,10     
AT_PKS.hmm,10</p>
</dd>
</dl>
<dl class="simple">
<dt>fasta_path<span class="classifier">string</span></dt><dd><p>Path of the created fasta file</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="pipeline_tools.protein_prediction.protein2bio">
<code class="descclassname">pipeline_tools.protein_prediction.</code><code class="descname">protein2bio</code><span class="sig-paren">(</span><em>HMMER_OUTPUT_PATH</em>, <em>GBK_PATH</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#pipeline_tools.protein_prediction.protein2bio" title="Permalink to this definition">¶</a></dt>
<dd><p>creates domain entrys in the gbk file, by converting hmmer3
search results into biopython seqfeatures, 
as the hmmer3 biopython parser does not store the 
original AS code, this needs to be extracted</p>
<dl class="simple">
<dt>HMMER_OUTPUT_PATH<span class="classifier">string</span></dt><dd><p>Path of the hmmer output files created by hmmer_wrapper</p>
</dd>
<dt>GBK_PATH<span class="classifier">string</span></dt><dd><p>Path of the gbk where the domains should be added</p>
</dd>
</dl>
<dl class="simple">
<dt>OFP_GBK2<span class="classifier">string</span></dt><dd><p>New GBK file with the added domains</p>
</dd>
</dl>
</dd></dl>

</div>
<div class="section" id="module-pipeline_tools.scoring_standalone_utils">
<span id="pipeline-tools-scoring-standalone-utils-module"></span><h2>pipeline_tools.scoring_standalone_utils module<a class="headerlink" href="#module-pipeline_tools.scoring_standalone_utils" title="Permalink to this headline">¶</a></h2>
<dl class="function">
<dt id="pipeline_tools.scoring_standalone_utils.create_dummy_cluster_df">
<code class="descclassname">pipeline_tools.scoring_standalone_utils.</code><code class="descname">create_dummy_cluster_df</code><span class="sig-paren">(</span><em>smiles_input=None</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#pipeline_tools.scoring_standalone_utils.create_dummy_cluster_df" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates a dummy cluster dataframe for the smiles input, 
which allows convenient passing on to the pipeline.</p>
<dl class="simple">
<dt>smiles_input<span class="classifier">str</span></dt><dd><p>smiles like CCCCC.CCC</p>
</dd>
</dl>
<dl class="simple">
<dt>cluster_df<span class="classifier">df</span></dt><dd><p>df which can be passed to get_functional_cluster_smiles_scoring</p>
</dd>
</dl>
</dd></dl>

</div>
<div class="section" id="module-pipeline_tools.structure_prediction">
<span id="pipeline-tools-structure-prediction-module"></span><h2>pipeline_tools.structure_prediction module<a class="headerlink" href="#module-pipeline_tools.structure_prediction" title="Permalink to this headline">¶</a></h2>
<dl class="function">
<dt id="pipeline_tools.structure_prediction.combine_substruct">
<code class="descclassname">pipeline_tools.structure_prediction.</code><code class="descname">combine_substruct</code><span class="sig-paren">(</span><em>df</em><span class="sig-paren">)</span><a class="headerlink" href="#pipeline_tools.structure_prediction.combine_substruct" title="Permalink to this definition">¶</a></dt>
<dd><p>Combines the substructures to a full molecule based on 
the substructures in a block</p>
<dl class="simple">
<dt>df<span class="classifier">pandas df</span></dt><dd><p>gbk_df</p>
</dd>
</dl>
<dl class="simple">
<dt>df<span class="classifier">pandas df</span></dt><dd><p>gbk_df with added substructs</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="pipeline_tools.structure_prediction.set_substructure_nrps">
<code class="descclassname">pipeline_tools.structure_prediction.</code><code class="descname">set_substructure_nrps</code><span class="sig-paren">(</span><em>df</em><span class="sig-paren">)</span><a class="headerlink" href="#pipeline_tools.structure_prediction.set_substructure_nrps" title="Permalink to this definition">¶</a></dt>
<dd><p>Create a substructure for the nrps module, based on 
the reduction profile and specificity</p>
<dl class="simple">
<dt>df<span class="classifier">pandas df</span></dt><dd><p>gbk_df</p>
</dd>
</dl>
<dl class="simple">
<dt>df<span class="classifier">pandas df</span></dt><dd><p>gbk_df with added substructs</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="pipeline_tools.structure_prediction.set_substructure_pks">
<code class="descclassname">pipeline_tools.structure_prediction.</code><code class="descname">set_substructure_pks</code><span class="sig-paren">(</span><em>df</em><span class="sig-paren">)</span><a class="headerlink" href="#pipeline_tools.structure_prediction.set_substructure_pks" title="Permalink to this definition">¶</a></dt>
<dd><p>Create a substructure for the pks module, based on 
the reduction profile and specificity</p>
<dl class="simple">
<dt>df<span class="classifier">pandas df</span></dt><dd><p>gbk_df</p>
</dd>
</dl>
<dl class="simple">
<dt>df<span class="classifier">pandas df</span></dt><dd><p>gbk_df with added substructs</p>
</dd>
</dl>
</dd></dl>

</div>
<div class="section" id="module-pipeline_tools">
<span id="module-contents"></span><h2>Module contents<a class="headerlink" href="#module-pipeline_tools" title="Permalink to this headline">¶</a></h2>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h3><a href="../index.html">Table of Contents</a></h3>
<p class="caption"><span class="caption-text">Contents:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../chapters/primer.html">Primer</a></li>
<li class="toctree-l1"><a class="reference internal" href="../chapters/genome_input.html">Genome input</a></li>
<li class="toctree-l1"><a class="reference internal" href="../chapters/prediction_algo.html">Prediction</a></li>
<li class="toctree-l1"><a class="reference internal" href="../chapters/genome_browser.html">Genome browser</a></li>
<li class="toctree-l1"><a class="reference internal" href="../chapters/database_screening.html">Database screening</a></li>
<li class="toctree-l1"><a class="reference internal" href="../chapters/update_log.html">Updates</a></li>
</ul>

<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="../search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="../py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li class="nav-item nav-item-0"><a href="../index.html">SeMPI 2.0 documentation</a> &#187;</li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &#169; Copyright 2019, Paul Zierep.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 2.0.1.
    </div>
  </body>
</html>