
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>Genome input &#8212; SeMPI 2.0 documentation</title>
    <link rel="stylesheet" href="../_static/nature.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../',
        VERSION:     '2.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true,
        SOURCELINK_SUFFIX: '.txt'
      };
    </script>
    <script type="text/javascript" src="../_static/jquery.js"></script>
    <script type="text/javascript" src="../_static/underscore.js"></script>
    <script type="text/javascript" src="../_static/doctools.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Prediction" href="prediction_algo.html" />
    <link rel="prev" title="Primer" href="primer.html" /> 
  </head>
  <body>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="prediction_algo.html" title="Prediction"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="primer.html" title="Primer"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="../index.html">SeMPI 2.0 documentation</a> &#187;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="genome-input">
<h1>Genome input<a class="headerlink" href="#genome-input" title="Permalink to this headline">¶</a></h1>
<div class="section" id="input-form">
<h2>Input form<a class="headerlink" href="#input-form" title="Permalink to this headline">¶</a></h2>
<div class="figure">
<a class="reference internal image-reference" href="../_images/input_form_01.svg"><img alt="input" src="../_images/input_form_01.svg" /></a>
</div>
<ol class="arabic simple">
<li>Upload a gene file (DNA) in FASTA or GeneBank format. Drop and drag is also supported.</li>
<li>Its easy to get confused, when multiple files are submitted to the server, therefore the jobs can be commented, which allows simple identification when the jobs are done. An example could be “interesting genome Number 3”. This field is optional.</li>
<li>The threshold which defines the maximum distance between functional modules allowing them to be part of the same cluster.</li>
<li>If modules are detected, which lack a functional AT domain, they might be modules using a so called trans-AT (<a class="reference external" href="https://pubs.rsc.org/en/content/articlelanding/2016/np/c5np00125k#!divAbstract">https://pubs.rsc.org/en/content/articlelanding/2016/np/c5np00125k#!divAbstract</a>). In SeMPI 2 their module specificity is set to undefined, but they are still used in the cluster creation process. If the box is unchecked, these modules are set to “not functional”.</li>
<li>Force the gene prediction algorithm. Independent of the input, the genes will be predicted using prodigal, based on the provided DNA. Otherwise the genes will only be predicted if they cannot be found in the input data. Pre-computed genes need to be assigned in a GeneBank file using the CDS tag (<a class="reference external" href="https://www.ncbi.nlm.nih.gov/Sitemap/samplerecord.html#CDSB">https://www.ncbi.nlm.nih.gov/Sitemap/samplerecord.html#CDSB</a>). As SeMPI 2.0 needs the stand side information of the gene (+ or -) for reasonable cluster predictions, files where the strand side is not assigned will automatically lead to a gene prediction.</li>
</ol>
<div class="figure">
<img alt="input" src="../_images/cds_strand.svg" /></div>
<p>(6) Force prodigal to use the meta (in new versions also called “Anonymous Mode”) option (predicting the genes based on an independent genome training dataset,
<a class="reference external" href="https://github.com/hyattpd/Prodigal/wiki/Gene-Prediction-Modes#anonymous-mode">https://github.com/hyattpd/Prodigal/wiki/Gene-Prediction-Modes#anonymous-mode</a>). In the normal settings, the meta option is only used, if the provided DNA is too short for accurate self-training.
(7) The screening algorithm uses fingerprints to detect similar compounds in the chosen data bases (DBs) (this gives already close matches). Of those matches the best N. compounds are used for a stricter screening process using most common substructures (MCS). The MCS algorithm can take a while to process, therefore it is reasonable to limit the number of compounds selected for this task (see also <a class="reference internal" href="database_screening.html#fp-screen"><span class="std std-ref">Fingerprint screening</span></a>).</p>
</div>
<div class="section" id="input-format">
<h2>Input format<a class="headerlink" href="#input-format" title="Permalink to this headline">¶</a></h2>
<p>SeMPI can work with genome data in FASTA or GenBank format.
An example file for both formats can be download from the start page.</p>
</div>
<div class="section" id="further-processing">
<h2>Further processing<a class="headerlink" href="#further-processing" title="Permalink to this headline">¶</a></h2>
<div class="figure" id="id1">
<a class="reference internal image-reference" href="../_images/input.svg"><img alt="input" src="../_images/input.svg" /></a>
<p class="caption"><span class="caption-text">If only DNA data is provided, the genes are predicted using prodigal. If the genes are already assigned (GenBank) SeMPI will try to parse the genes and use them for further analysis. SeMPI can parse multiple records per file, it will create one output for each record.</span></p>
</div>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <h4>Previous topic</h4>
  <p class="topless"><a href="primer.html"
                        title="previous chapter">Primer</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="prediction_algo.html"
                        title="next chapter">Prediction</a></p>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="../search.html" method="get">
      <div><input type="text" name="q" /></div>
      <div><input type="submit" value="Go" /></div>
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="prediction_algo.html" title="Prediction"
             >next</a> |</li>
        <li class="right" >
          <a href="primer.html" title="Primer"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="../index.html">SeMPI 2.0 documentation</a> &#187;</li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &#169; Copyright 2019, Paul Zierep.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.6.6.
    </div>
  </body>
</html>