
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Prediction &#8212; SeMPI 2.0 documentation</title>
    <link rel="stylesheet" href="../_static/nature.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <script id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/language_data.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Genome browser" href="genome_browser.html" />
    <link rel="prev" title="Generated project files" href="output_files.html" /> 
  </head><body>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="genome_browser.html" title="Genome browser"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="output_files.html" title="Generated project files"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="../index.html">SeMPI 2.0 documentation</a> &#187;</li>
        <li class="nav-item nav-item-this"><a href="">Prediction</a></li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="prediction">
<h1>Prediction<a class="headerlink" href="#prediction" title="Permalink to this headline">¶</a></h1>
<p>This is the work flow of the gene cluster detection and scaffold pipeline.</p>
<div class="section" id="gene-prediction">
<h2>Gene prediction<a class="headerlink" href="#gene-prediction" title="Permalink to this headline">¶</a></h2>
<p>If only DNA data is provided, the genes are
predicted using <a class="reference external" href="https://github.com/hyattpd/Prodigal">Prodigal</a>.
If the genes are already assigned, SeMPI can try to parse the genes and
use them for further analysis.</p>
</div>
<div class="section" id="domain-detection">
<span id="id1"></span><h2>Domain detection<a class="headerlink" href="#domain-detection" title="Permalink to this headline">¶</a></h2>
<p>The detection of relevant domains in the predicted proteins is performed using profile hidden Markov models (profile-HMMs)
created with HMMER 3.0. The detected domains are shown in <a class="reference internal" href="#phmms"><span class="std std-numref">Table 1</span></a>. Since the profile-HMMs are derived and modified from the MIBiG 2 database, the nomenclature is mainly based on the antiSMAHS definition as described in their <a class="reference external" href="https://docs.antismash.secondarymetabolites.org/modules/nrps_pks_domains/">documentation</a>.</p>
<span id="phmms"></span><table class="colwidths-given docutils align-default" id="id2">
<caption><span class="caption-number">Table 1 </span><span class="caption-text">HMM profiles used for domain detection. The F1-scores and thresholds are based on the detection performance of the profile-HMMs on the <a class="reference external" href="https://mibig.secondarymetabolites.org/">MIBiG 2.0 database</a>.</span><a class="headerlink" href="#id2" title="Permalink to this table">¶</a></caption>
<colgroup>
<col style="width: 20%" />
<col style="width: 40%" />
<col style="width: 20%" />
<col style="width: 20%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Domain</p></th>
<th class="head"><p>Full name</p></th>
<th class="head"><p>F1-score</p></th>
<th class="head"><p>HMM detection threshold (domT)</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>ACP</p></td>
<td><p>Acyl carrier protein</p></td>
<td><p>0.96</p></td>
<td><p>13.9</p></td>
</tr>
<tr class="row-odd"><td><p>AT</p></td>
<td><p>Acyltransferase</p></td>
<td><p>1</p></td>
<td><p>47.4</p></td>
</tr>
<tr class="row-even"><td><p>A</p></td>
<td><p>Adenylation domain</p></td>
<td><p>1</p></td>
<td><p>19.6</p></td>
</tr>
<tr class="row-odd"><td><p>CAL</p></td>
<td><p>Coenzyme A ligase</p></td>
<td><p>0.94</p></td>
<td><p>221.8</p></td>
</tr>
<tr class="row-even"><td><p>C</p></td>
<td><p>Condensation domain</p></td>
<td><p>1</p></td>
<td><p>25.9</p></td>
</tr>
<tr class="row-odd"><td><p>DH2</p></td>
<td><p>Dehydratase</p></td>
<td><p>0.98</p></td>
<td><p>33.2</p></td>
</tr>
<tr class="row-even"><td><p>DH</p></td>
<td><p>Dehydratase</p></td>
<td><p>0.99</p></td>
<td><p>31.1</p></td>
</tr>
<tr class="row-odd"><td><p>DHt</p></td>
<td><p>Dehydratase</p></td>
<td><p>0.88</p></td>
<td><p>44.3</p></td>
</tr>
<tr class="row-even"><td><p>ER</p></td>
<td><p>Enoylreductase</p></td>
<td><p>1</p></td>
<td><p>68.8</p></td>
</tr>
<tr class="row-odd"><td><p>E</p></td>
<td><p>Epimerization domain</p></td>
<td><p>1</p></td>
<td><p>60.4</p></td>
</tr>
<tr class="row-even"><td><p>KR</p></td>
<td><p>Ketoreductase</p></td>
<td><p>1</p></td>
<td><p>20</p></td>
</tr>
<tr class="row-odd"><td><p>KS</p></td>
<td><p>Keto-synthase</p></td>
<td><p>1</p></td>
<td><p>72</p></td>
</tr>
<tr class="row-even"><td><p>PCP</p></td>
<td><p>Peptide carrier protein</p></td>
<td><p>0.97</p></td>
<td><p>21.9</p></td>
</tr>
<tr class="row-odd"><td><p>TD</p></td>
<td><p>Reductive Thioesterase</p></td>
<td><p>1</p></td>
<td><p>42.1</p></td>
</tr>
<tr class="row-even"><td><p>TE</p></td>
<td><p>Thioesterase</p></td>
<td><p>1</p></td>
<td><p>36.5</p></td>
</tr>
<tr class="row-odd"><td><p>bACP</p></td>
<td><p>β-branching acyl carrier protein</p></td>
<td><p>0.59</p></td>
<td><p>28.9</p></td>
</tr>
<tr class="row-even"><td><p>cMT</p></td>
<td><p>C-Methyltransferase</p></td>
<td><p>1</p></td>
<td><p>104.7</p></td>
</tr>
<tr class="row-odd"><td><p>nMT</p></td>
<td><p>N-Methyltransferase</p></td>
<td><p>0.99</p></td>
<td><p>42.7</p></td>
</tr>
<tr class="row-even"><td><p>oMT</p></td>
<td><p>O-Methyltransferase</p></td>
<td><p>0.99</p></td>
<td><p>70.6</p></td>
</tr>
<tr class="row-odd"><td><p>tATd</p></td>
<td><p>Trans-acyltransferase docking domain</p></td>
<td><p>0.98</p></td>
<td><p>50</p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="gene-cluster-curation">
<span id="bgc-curration"></span><h2>Gene cluster curation<a class="headerlink" href="#gene-cluster-curation" title="Permalink to this headline">¶</a></h2>
<p>After the protein detection the genome is converted into a DataFrame <a class="reference external" href="https://pandas.pydata.org/">pandas</a>, which simplifies the subsequent operations. Additional, pandas allows to apply vectorized functions, which speeds up the prediction pipeline. A mysql dump of the DataFrames is provided with the final output.</p>
<p>The DataFrame is curated and modified in order to prepare the BGC module detection (see Example <a class="reference internal" href="#df-processing"><span class="std std-numref">Fig. 8</span></a>).</p>
<ol class="loweralpha simple">
<li><p>Close genes (the threshold can be set in the options of the <a class="reference internal" href="genome_input.html#upload-form"><span class="std std-ref">Upload form</span></a>) which encode proteins on the same strand are combined into blocks. A detailed observation of the MIBiG annotated NRP and PK clusters showed, that for this condition the co-linearity principle applies for the very most cases.</p></li>
<li><p>The domains are ordered based on the occurrence in the genome (id), but for module assignment it is more useful to order the domains based on the occurrence in the block, therefore an additional index is assigned based on the block order (Corr. id).</p></li>
<li><p>In very rare cases the detected protein domains overlap. This can be the case especially for very short sequences (for example the ACP domain). If the domains overlap to more then 20% the domain with the lower bitscore (see <a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3125773/">HMMER</a>) is removed.</p></li>
<li><p>Sometimes the HMM algorithm detects two domains in succession instead of one domain, these domains are automatically joined to one domain.</p></li>
</ol>
<div class="figure align-default" id="id3">
<span id="df-processing"></span><a class="reference internal image-reference" href="../_images/df_processing.svg"><img alt="prediction" src="../_images/df_processing.svg" /></a>
<p class="caption"><span class="caption-number">Fig. 8 </span><span class="caption-text">Curation steps: (a) Block generation, (b) Domain reordering, (c) Domain overlap removal, (d) Domain duplication joining.</span><a class="headerlink" href="#id3" title="Permalink to this image">¶</a></p>
</div>
</div>
<div class="section" id="module-detection">
<h2>Module detection<a class="headerlink" href="#module-detection" title="Permalink to this headline">¶</a></h2>
<p>SeMPI v2 detects functional PKS and NRPS modules based on a in-depth analysis
of known gene clusters stored in the MIBiG 2.0.
The modules can be be observed either in the tabular
representation (see Example <a class="reference internal" href="#mods1"><span class="std std-numref">Fig. 9</span></a>) of the domains of a gene cluster or
in the interactive cluster browser (see Example <a class="reference internal" href="#mods2"><span class="std std-numref">Fig. 10</span></a>).</p>
<div class="figure align-default" id="id4">
<span id="mods1"></span><img alt="mods1" src="../_images/mods_01.png" />
<p class="caption"><span class="caption-number">Fig. 9 </span><span class="caption-text">Tabular domain representation of a gene cluster with modules highlighted in red.</span><a class="headerlink" href="#id4" title="Permalink to this image">¶</a></p>
</div>
<div class="figure align-default" id="id5">
<span id="mods2"></span><img alt="mods2" src="../_images/mods_02.png" />
<p class="caption"><span class="caption-number">Fig. 10 </span><span class="caption-text">Cluster browser representation of a gene cluster with modules highlighted in red.</span><a class="headerlink" href="#id5" title="Permalink to this image">¶</a></p>
</div>
</div>
<div class="section" id="assembly-logic-for-disjointed-gene-clusters">
<h2>Assembly logic for disjointed gene clusters<a class="headerlink" href="#assembly-logic-for-disjointed-gene-clusters" title="Permalink to this headline">¶</a></h2>
<p>If the domains are not organized in a single consecutive block an module ordering algorithm is applied.
In most cases (&lt;= 3 blocks) the assignment of starting modules to the beginning and terminal modules to the end of an gene cluster
is sufficient for a correct module order.
For gene clusters with more than 3 blocks the modules are ordered based on comparison of the
gene cluster assembly with the set-up of more than 250 reference gene clusters where the module order is known.</p>
</div>
<div class="section" id="postsynthetic-modifications">
<h2>Postsynthetic modifications<a class="headerlink" href="#postsynthetic-modifications" title="Permalink to this headline">¶</a></h2>
<p>In order to detect postsynthetic modifications additional <a class="reference external" href="http://pfam.xfam.org/">Pfam domains</a> are detected in proximity of the gene cluster.
The Pfam domains are correlated to specific postsynthetic modifications as shown in <a class="reference internal" href="#pfams"><span class="std std-numref">Table 2</span></a>.</p>
<span id="pfams"></span><table class="colwidths-given docutils align-default" id="id6">
<caption><span class="caption-number">Table 2 </span><span class="caption-text">Detected Pfam domains correlated to common postsynthetic modifications.</span><a class="headerlink" href="#id6" title="Permalink to this table">¶</a></caption>
<colgroup>
<col style="width: 20%" />
<col style="width: 20%" />
<col style="width: 40%" />
<col style="width: 20%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Postsynthetic modification</p></th>
<th class="head"><p>Pfam ID</p></th>
<th class="head"><p>Description</p></th>
<th class="head"><p>Correlation</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>Glyco</p></td>
<td><p>PF00908.16</p></td>
<td><p>dTDP-4-dehydrorhamnose 3,5-epimerase</p></td>
<td><p>0.67</p></td>
</tr>
<tr class="row-odd"><td><p>Glyco</p></td>
<td><p>PF01370.20</p></td>
<td><p>NAD dependent epimerase/dehydratase family</p></td>
<td><p>0.64</p></td>
</tr>
<tr class="row-even"><td><p>Glyco</p></td>
<td><p>PF03559.13</p></td>
<td><p>NDP-hexose 2,3-dehydratase</p></td>
<td><p>0.61</p></td>
</tr>
<tr class="row-odd"><td><p>Glyco</p></td>
<td><p>PF00201.17</p></td>
<td><p>UDP-glucoronosyl and UDP-glucosyl transferase</p></td>
<td><p>0.59</p></td>
</tr>
<tr class="row-even"><td><p>Glyco</p></td>
<td><p>PF03033.19</p></td>
<td><p>Glycosyltransferase family 28 N-terminal domain</p></td>
<td><p>0.56</p></td>
</tr>
<tr class="row-odd"><td><p>Glyco</p></td>
<td><p>PF01041.16</p></td>
<td><p>DegT/DnrJ/EryC1/StrS aminotransferase family</p></td>
<td><p>0.52</p></td>
</tr>
<tr class="row-even"><td><p>Glyco</p></td>
<td><p>PF08421.10</p></td>
<td><p>Putative zinc binding domain</p></td>
<td><p>0.47</p></td>
</tr>
<tr class="row-odd"><td><p>Glyco</p></td>
<td><p>PF16363.4</p></td>
<td><p>GDP-mannose 4,6 dehydratase</p></td>
<td><p>0.44</p></td>
</tr>
<tr class="row-even"><td><p>Glyco</p></td>
<td><p>PF04101.15</p></td>
<td><p>Glycosyltransferase family 28 C-terminal domain</p></td>
<td><p>0.28</p></td>
</tr>
<tr class="row-odd"><td><p>Glyco</p></td>
<td><p>PF01075.16</p></td>
<td><p>Glycosyltransferase family 9 (heptosyltransferase)</p></td>
<td><p>0.21</p></td>
</tr>
<tr class="row-even"><td><p>Glyco</p></td>
<td><p>PF00728.21</p></td>
<td><p>Glycosyl hydrolase family 20, catalytic domain</p></td>
<td><p>0.18</p></td>
</tr>
<tr class="row-odd"><td><p>Glyco</p></td>
<td><p>PF02838.14</p></td>
<td><p>Glycosyl hydrolase family 20, domain 2</p></td>
<td><p>0.18</p></td>
</tr>
<tr class="row-even"><td><p>Glyco</p></td>
<td><p>PF01915.21</p></td>
<td><p>Glycosyl hydrolase family 3 C-terminal domain</p></td>
<td><p>0.13</p></td>
</tr>
<tr class="row-odd"><td><p>Glyco</p></td>
<td><p>PF00933.20</p></td>
<td><p>Glycosyl hydrolase family 3 N terminal domain</p></td>
<td><p>0.13</p></td>
</tr>
<tr class="row-even"><td><p>Glyco</p></td>
<td><p>PF14885.5</p></td>
<td><p>Hypothetical glycosyl hydrolase family 15</p></td>
<td><p>0.1</p></td>
</tr>
<tr class="row-odd"><td><p>Cl</p></td>
<td><p>PF04820.13</p></td>
<td><p>Tryptophan halogenase</p></td>
<td><p>0.66</p></td>
</tr>
<tr class="row-even"><td><p>Cl</p></td>
<td><p>PF00999.20</p></td>
<td><p>Sodium/hydrogen exchanger family</p></td>
<td><p>0.51</p></td>
</tr>
<tr class="row-odd"><td><p>Sphingo</p></td>
<td><p>PF12680.6</p></td>
<td><p>SnoaL-like domain</p></td>
<td><p>0.61</p></td>
</tr>
<tr class="row-even"><td><p>Sphingo</p></td>
<td><p>PF00890.23</p></td>
<td><p>FAD binding domain</p></td>
<td><p>0.54</p></td>
</tr>
<tr class="row-odd"><td><p>SS</p></td>
<td><p>PF07992.13</p></td>
<td><p>Pyridine nucleotide-disulphide oxidoreductase</p></td>
<td><p>0.46</p></td>
</tr>
<tr class="row-even"><td><p>NO2</p></td>
<td><p>PF01678.18</p></td>
<td><p>Diaminopimelate epimerase</p></td>
<td><p>0.67</p></td>
</tr>
<tr class="row-odd"><td><p>6-Ring</p></td>
<td><p>PF16197.4</p></td>
<td><p>Ketoacyl-synthetase C-terminal extension</p></td>
<td><p>0.53</p></td>
</tr>
<tr class="row-even"><td><p>6-Ring</p></td>
<td><p>PF02801.21</p></td>
<td><p>Beta-ketoacyl synthase, C-terminal domain</p></td>
<td><p>0.52</p></td>
</tr>
<tr class="row-odd"><td><p>6-Ring</p></td>
<td><p>PF08990.10</p></td>
<td><p>Erythronolide synthase docking</p></td>
<td><p>0.51</p></td>
</tr>
<tr class="row-even"><td><p>6-Ring</p></td>
<td><p>PF00743.18</p></td>
<td><p>Flavin-binding monooxygenase-like</p></td>
<td><p>0.46</p></td>
</tr>
<tr class="row-odd"><td><p>6-Ring</p></td>
<td><p>PF13377.5</p></td>
<td><p>Periplasmic binding protein-like domain</p></td>
<td><p>0.23</p></td>
</tr>
<tr class="row-even"><td><p>6-Ring</p></td>
<td><p>PF13191.5</p></td>
<td><p>AAA ATPase domain</p></td>
<td><p>0.23</p></td>
</tr>
<tr class="row-odd"><td><p>5-Ring</p></td>
<td><p>PF12680.6</p></td>
<td><p>SnoaL-like domain</p></td>
<td><p>0.56</p></td>
</tr>
<tr class="row-even"><td><p>5-Ring</p></td>
<td><p>PF00890.23</p></td>
<td><p>FAD binding domain</p></td>
<td><p>0.53</p></td>
</tr>
<tr class="row-odd"><td><p>5-Ring</p></td>
<td><p>PF01551.21</p></td>
<td><p>Peptidase family M23</p></td>
<td><p>0.46</p></td>
</tr>
<tr class="row-even"><td><p>5-Ring</p></td>
<td><p>PF08990.10</p></td>
<td><p>Erythronolide synthase docking</p></td>
<td><p>0.45</p></td>
</tr>
<tr class="row-odd"><td><p>5-Ring</p></td>
<td><p>PF00486.27</p></td>
<td><p>Transcriptional regulatory protein, C terminal</p></td>
<td><p>0.26</p></td>
</tr>
<tr class="row-even"><td><p>5-Ring</p></td>
<td><p>PF16197.4</p></td>
<td><p>Ketoacyl-synthetase C-terminal extension</p></td>
<td><p>0.26</p></td>
</tr>
<tr class="row-odd"><td><p>5-Ring</p></td>
<td><p>PF00109.25</p></td>
<td><p>Beta-ketoacyl synthase, N-terminal domain</p></td>
<td><p>0.25</p></td>
</tr>
</tbody>
</table>
<p>Based on the detected domains the postsynthetic modifications are predicted using regression models.
The predicted numbers of postsynthetic modifications are shown together with the predicted scaffold as shown in <a class="reference internal" href="#psm"><span class="std std-numref">Fig. 11</span></a>.</p>
<div class="figure align-default" id="id7">
<span id="psm"></span><a class="reference internal image-reference" href="../_images/psms_example.jpg"><img alt="psm example" src="../_images/psms_example.jpg" style="width: 714.6px; height: 288.0px;" /></a>
<p class="caption"><span class="caption-number">Fig. 11 </span><span class="caption-text">Example of an PK scaffold prediction with indicated halogen (Cl) postsynthetic modification.</span><a class="headerlink" href="#id7" title="Permalink to this image">¶</a></p>
</div>
</div>
<div class="section" id="scaffold-generation">
<h2>Scaffold generation<a class="headerlink" href="#scaffold-generation" title="Permalink to this headline">¶</a></h2>
<p>The scaffolds are generated by joining of each building block of a gene cluster.
If the modules of a cluster cannot be joined multiple blocks are generated (see Example <a class="reference internal" href="#scaff1"><span class="std std-numref">Fig. 12</span></a>). The database screening queries all blocks in the target
molecules (see Example <a class="reference internal" href="#scaff2"><span class="std std-numref">Fig. 13</span></a>).
The example is taken from <a class="reference external" href="http://sempi.pharmazie.uni-freiburg.de/results/21839">BGC0000317</a>.
The multiple blocks scaffoled scoring algorithm is further described in <a class="reference internal" href="database_screening.html#mcs-rank"><span class="std std-ref">Maximum Common Substructure Ranking</span></a>.</p>
<div class="figure align-default" id="id8">
<span id="scaff1"></span><img alt="mods1" src="../_images/scaff1.jpg" />
<p class="caption"><span class="caption-number">Fig. 12 </span><span class="caption-text">Two blocks predicted for one cluster.</span><a class="headerlink" href="#id8" title="Permalink to this image">¶</a></p>
</div>
<div class="figure align-default" id="id9">
<span id="scaff2"></span><img alt="mods2" src="../_images/scaff2.jpg" />
<p class="caption"><span class="caption-number">Fig. 13 </span><span class="caption-text">Both blocks are screened in each of the target compounds.</span><a class="headerlink" href="#id9" title="Permalink to this image">¶</a></p>
</div>
</div>
</div>


            <div class="clearer"></div>
          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h3><a href="../index.html">Table of Contents</a></h3>
<p class="caption"><span class="caption-text">Description of the SeMPI 2.0 web-server</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="primer.html">Primer</a></li>
<li class="toctree-l1"><a class="reference internal" href="genome_input.html">Genome input</a></li>
<li class="toctree-l1"><a class="reference internal" href="result_page.html">Result page</a></li>
<li class="toctree-l1"><a class="reference internal" href="output_files.html">Generated project files</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Prediction</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#gene-prediction">Gene prediction</a></li>
<li class="toctree-l2"><a class="reference internal" href="#domain-detection">Domain detection</a></li>
<li class="toctree-l2"><a class="reference internal" href="#gene-cluster-curation">Gene cluster curation</a></li>
<li class="toctree-l2"><a class="reference internal" href="#module-detection">Module detection</a></li>
<li class="toctree-l2"><a class="reference internal" href="#assembly-logic-for-disjointed-gene-clusters">Assembly logic for disjointed gene clusters</a></li>
<li class="toctree-l2"><a class="reference internal" href="#postsynthetic-modifications">Postsynthetic modifications</a></li>
<li class="toctree-l2"><a class="reference internal" href="#scaffold-generation">Scaffold generation</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="genome_browser.html">Genome browser</a></li>
<li class="toctree-l1"><a class="reference internal" href="database_screening.html">Database screening</a></li>
<li class="toctree-l1"><a class="reference internal" href="preprocessed_database.html">Preprocessed Database</a></li>
</ul>
<p class="caption"><span class="caption-text">Update log</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="update_log.html">Updates</a></li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="output_files.html"
                        title="previous chapter">Generated project files</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="genome_browser.html"
                        title="next chapter">Genome browser</a></p>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="../search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" />
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="genome_browser.html" title="Genome browser"
             >next</a> |</li>
        <li class="right" >
          <a href="output_files.html" title="Generated project files"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="../index.html">SeMPI 2.0 documentation</a> &#187;</li>
        <li class="nav-item nav-item-this"><a href="">Prediction</a></li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &#169; Copyright 2019, Paul Zierep.
      Created using <a href="https://www.sphinx-doc.org/">Sphinx</a> 3.1.2.
    </div>
  </body>
</html>