
<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta charset="utf-8" />
    <title>Prediction &#8212; SeMPI 2.0 documentation</title>
    <link rel="stylesheet" href="../_static/nature.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script type="text/javascript" src="../_static/jquery.js"></script>
    <script type="text/javascript" src="../_static/underscore.js"></script>
    <script type="text/javascript" src="../_static/doctools.js"></script>
    <script type="text/javascript" src="../_static/language_data.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Genome browser" href="genome_browser.html" />
    <link rel="prev" title="Genome input" href="genome_input.html" /> 
  </head><body>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="genome_browser.html" title="Genome browser"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="genome_input.html" title="Genome input"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="../index.html">SeMPI 2.0 documentation</a> &#187;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="prediction">
<h1>Prediction<a class="headerlink" href="#prediction" title="Permalink to this headline">¶</a></h1>
<div class="section" id="overview">
<h2>Overview<a class="headerlink" href="#overview" title="Permalink to this headline">¶</a></h2>
<div class="figure align-center" id="id1">
<a class="reference internal image-reference" href="../_images/prediction.svg"><img alt="prediction" src="../_images/prediction.svg" /></a>
<p class="caption"><span class="caption-number">Fig. 4 </span><span class="caption-text">Flow chart of the prediction pipeline.</span><a class="headerlink" href="#id1" title="Permalink to this image">¶</a></p>
</div>
</div>
<div class="section" id="curration-1">
<h2>Curration (1)<a class="headerlink" href="#curration-1" title="Permalink to this headline">¶</a></h2>
<div class="figure align-center" id="id2">
<span id="df-processing"></span><a class="reference internal image-reference" href="../_images/df_processing.svg"><img alt="prediction" src="../_images/df_processing.svg" /></a>
<p class="caption"><span class="caption-number">Fig. 5 </span><span class="caption-text">Curration steps: (a) Block generation, (b) Domain reordering, (c) Domain overlap removal, (d) Domain doublication joining.</span><a class="headerlink" href="#id2" title="Permalink to this image">¶</a></p>
</div>
<p>After the protein detection the genome is converted into a data frame (pandas), which simplifies the subsequent operations. Additional, pandas allows to apply vectorized functions, which speeds up the prediction pipeline. A mysql dump of the data frame is provided with the final output.</p>
<p>The data frame is curated and modified in order to prepare the BGC module detection (see Example <a class="reference internal" href="#df-processing"><span class="std std-numref">Fig. 5</span></a>).
(a) Close genes (&lt;= 5000bp) which encode proteins on the same strand are combined into blocks. A detailed observation of the MIBiG annotated NRPS and PKS clusters showed, that for this condition the co-linearity principle applies for the very most cases.
(b) The domains are ordered based on the occurrence in the genome, but for module assignment it is more useful to order the domains based on the occurrence in the block, therefore an additional index is assigned based on the block order.
(c) In very rare cases the detected protein domains overlap. This can be the case especially for very short sequences (for example the ACP domain). If the domains overlap to more then 20% the domain with the lower bitscore is removed.
(d) Sometimes the HMM algorithm detects two domains behind each other instead of one domain, these domains are automatically joined to one domain.</p>
</div>
<div class="section" id="protein-prediction">
<h2>Protein prediction<a class="headerlink" href="#protein-prediction" title="Permalink to this headline">¶</a></h2>
<p>The detection of relevant proteins in the genes is performed using profile hidden Markov models (pHMMs) created with HMMER 3.0. This step is common to most prediction tools at present. But as HMMER 3.0 is the state-of-the-art for sensitive sequence homologue detection, there is no reason to reinvent the wheel. Nevertheless, SeMPI uses its own profiles created with in-house sequence databases. The following proteins are detected:</p>
<table class="colwidths-given docutils align-center" id="id3">
<caption><span class="caption-number">Table 1 </span><span class="caption-text">HMM profiles used for domain detection</span><a class="headerlink" href="#id3" title="Permalink to this table">¶</a></caption>
<colgroup>
<col style="width: 25%" />
<col style="width: 25%" />
<col style="width: 50%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Full name</p></th>
<th class="head"><p>Abbrev.</p></th>
<th class="head"><p>Remark</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>acyl carrier protein / peptidyl carrier proteins</p></td>
<td><p>ACP/PCP</p></td>
<td><p>ACP and PCP are detected using the same profile, but are subsequently assigned depending on the module where they occure.</p></td>
</tr>
<tr class="row-odd"><td><p>Acetyltransferase</p></td>
<td><p>AT</p></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Keto-synthase</p></td>
<td><p>KS</p></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Dehydratase</p></td>
<td><p>DH</p></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Enoylreductase</p></td>
<td><p>ER</p></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Ketoreductase</p></td>
<td><p>KR</p></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Methyltransferase</p></td>
<td><p>MT</p></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Thioesterase</p></td>
<td><p>TE</p></td>
<td></td>
</tr>
<tr class="row-even"><td><p>Condensation domain</p></td>
<td><p>C</p></td>
<td></td>
</tr>
<tr class="row-odd"><td><p>Epimerization domain</p></td>
<td><p>E</p></td>
<td><p>This domain is detected with the same pHMM as the condensation domain, but subsequently changed when positioned behind another condensation domain.</p></td>
</tr>
<tr class="row-even"><td><p>Adenylation domain</p></td>
<td><p>A</p></td>
<td></td>
</tr>
</tbody>
</table>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h3><a href="../index.html">Table of Contents</a></h3>
<p class="caption"><span class="caption-text">Description of the SeMPI 2.0 web-server</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="primer.html">Primer</a></li>
<li class="toctree-l1"><a class="reference internal" href="genome_input.html">Genome input</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Prediction</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#overview">Overview</a></li>
<li class="toctree-l2"><a class="reference internal" href="#curration-1">Curration (1)</a></li>
<li class="toctree-l2"><a class="reference internal" href="#protein-prediction">Protein prediction</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="genome_browser.html">Genome browser</a></li>
<li class="toctree-l1"><a class="reference internal" href="database_screening.html">Database screening</a></li>
</ul>
<p class="caption"><span class="caption-text">Update log</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="update_log.html">Updates</a></li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="genome_input.html"
                        title="previous chapter">Genome input</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="genome_browser.html"
                        title="next chapter">Genome browser</a></p>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="../search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="genome_browser.html" title="Genome browser"
             >next</a> |</li>
        <li class="right" >
          <a href="genome_input.html" title="Genome input"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="../index.html">SeMPI 2.0 documentation</a> &#187;</li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &#169; Copyright 2019, Paul Zierep.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 2.0.1.
    </div>
  </body>
</html>