
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Database screening &#8212; SeMPI 2.0 documentation</title>
    <link rel="stylesheet" href="../_static/nature.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <script id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/language_data.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Updates" href="update_log.html" />
    <link rel="prev" title="Genome browser" href="genome_browser.html" /> 
  </head><body>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="update_log.html" title="Updates"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="genome_browser.html" title="Genome browser"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="../index.html">SeMPI 2.0 documentation</a> &#187;</li>
        <li class="nav-item nav-item-this"><a href="">Database screening</a></li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="database-screening">
<h1>Database screening<a class="headerlink" href="#database-screening" title="Permalink to this headline">¶</a></h1>
<p>SeMPI allows to screen publicly available natural compound databases (DBs).
The query can be the predicted scaffolds of a cluster, a user defined modification of
scaffolds or a totally user defined set of molecules.</p>
<p>The DB screening is not limited to one scaffold. SeMPI can predict multiple fragment-scaffolds for one cluster (in case the order of the blocks cannot be determined). The screening therefore allows to submit multiple scaffolds for one query.</p>
<p>Additionally to the pre-computed DBs, the user can upload custom defined DBs in SDF or SMILES format.</p>
<p>The database screening can also be performed for user defined molecules and molecule sets,
entirely without preliminarily prediction.
Just follow the “scaffold upload” button
<a class="reference external" href="http://132.230.102.133/scaffold_upload">Scaffold upload</a>
and submit any molecule query to the screening algorithm.</p>
<div class="section" id="use-cases">
<h2>Use cases<a class="headerlink" href="#use-cases" title="Permalink to this headline">¶</a></h2>
<p>The database (DB) screening allows users:</p>
<ol class="arabic simple">
<li><p>to estimate the novelty of their gene clusters (a scaffold which does not match similar scaffolds in known databases, is very likely a novel cluster)</p></li>
<li><p>the identification of possible post-modifications, which cannot be detected by prediction algorithms yet</p></li>
<li><p>the identification of similar clusters (based on the product)</p></li>
<li><p>the matching of custom defined molecules in a genome (using the scaffold upload tool)</p></li>
</ol>
</div>
<div class="section" id="algorithm">
<h2>Algorithm<a class="headerlink" href="#algorithm" title="Permalink to this headline">¶</a></h2>
<div class="figure align-default" id="id2">
<span id="db-screen"></span><a class="reference internal image-reference" href="../_images/scaffold_screening.svg"><img alt="scaffold_screening" src="../_images/scaffold_screening.svg" /></a>
<p class="caption"><span class="caption-number">Fig. 6 </span><span class="caption-text">Flow chart of the DB screening pipeline.</span><a class="headerlink" href="#id2" title="Permalink to this image">¶</a></p>
</div>
<p>The database screening uses a two step approach to identify similar compounds from public DBs (see <a class="reference internal" href="#db-screen"><span class="std std-numref">Fig. 6</span></a>). First, a pre-selection of possible matches are identified using Morgan fingerprints (see <a class="reference internal" href="#fp-screen"><span class="std std-ref">Fingerprint screening</span></a>). Second, the pre-selection is ranked using a Maximum Common Substructure Approach (see <a class="reference internal" href="#mcs-rank"><span class="std std-ref">Maximum Common Substructure Ranking</span></a>). The scaffolds can be modified before submission (see <a class="reference internal" href="#scaffold-mod"><span class="std std-ref">Scaffold modification</span></a>).</p>
</div>
<div class="section" id="scaffold-modification">
<span id="scaffold-mod"></span><h2>Scaffold modification<a class="headerlink" href="#scaffold-modification" title="Permalink to this headline">¶</a></h2>
<div class="figure align-default" id="id3">
<a class="reference internal image-reference" href="../_images/scaffold_mod_01.png"><img alt="scaffold_mod_01" src="../_images/scaffold_mod_01.png" style="width: 767.5px; height: 365.0px;" /></a>
<p class="caption"><span class="caption-number">Fig. 7 </span><span class="caption-text">The predicted scaffolds can be modified before submission to the database screening, using the scaffold input button.</span><a class="headerlink" href="#id3" title="Permalink to this image">¶</a></p>
</div>
<div class="figure align-default" id="id4">
<a class="reference internal image-reference" href="../_images/scaffold_mod_02.png"><img alt="scaffold_mod_02" src="../_images/scaffold_mod_02.png" style="width: 431.5px; height: 432.0px;" /></a>
<p class="caption"><span class="caption-number">Fig. 8 </span><span class="caption-text">The JSME editor allows all kinds of modifications of the predicted scaffold.</span><a class="headerlink" href="#id4" title="Permalink to this image">¶</a></p>
</div>
<div class="figure align-default" id="id5">
<a class="reference internal image-reference" href="../_images/scaffold_mod_03.png"><img alt="scaffold_mod_03" src="../_images/scaffold_mod_03.png" style="width: 392.5px; height: 417.5px;" /></a>
<p class="caption"><span class="caption-number">Fig. 9 </span><span class="caption-text">Additional building blocks or post-modifications can be added to the original prediction.</span><a class="headerlink" href="#id5" title="Permalink to this image">¶</a></p>
</div>
<p>The scaffold modification can be performed using the <a class="reference external" href="https://peter-ertl.com/jsme/2013_03/help.html">JSME Molecule Editor</a>, a very powerful editor, which allows to add new molecules and modify the
original prediction.</p>
</div>
<div class="section" id="custom-db-input-format">
<span id="custom-db"></span><h2>Custom DB input format<a class="headerlink" href="#custom-db-input-format" title="Permalink to this headline">¶</a></h2>
<p>The custom DB choice allows to upload a file with user defined target molecules which can be screened additionally to the default DB choices. It is also possible to screen only a user defined DB, just untick
all the default DB choices before submission.</p>
<p>The format for the custom DB must be either SDF or SMILES format.
Examples of both input types can be downloaded from the <a class="reference external" href="http://132.230.102.133/scaffold_upload">Scaffold upload</a> page, at the bottom. These files simply contain a benzene and phenol structure which be used to
test the scaffold screening.</p>
<p>One simple way to understand the screening pipeline could be to download one of the example input files. And screen them for a custom drawn structure.</p>
</div>
<div class="section" id="fingerprint-screening">
<span id="fp-screen"></span><h2>Fingerprint screening<a class="headerlink" href="#fingerprint-screening" title="Permalink to this headline">¶</a></h2>
<p>The selected DBs are screened using molecular fingerprints (FP). For the screening the rdkit implementation
of Morgan FPs (Circular FPs) were chosen (<a class="reference external" href="https://www.rdkit.org/docs/GettingStartedInPython.html#fingerprinting-and-molecular-similarity">Morgan Fingerprints</a>).
Specifically a Morgan FB with radius 3 and features is implemented.</p>
<p>In many cases the scaffolds of PKs and NRPs contain the same building blocks multiple times.
This property can not
be accurately represented by bit fingerprints. Multiple identical building blocks set the same bits (and
bits are only set ones), which leads to similar fingerprints for scaffolds with different amounts of identical building blocks.
Therefore, the FPs are implemented as count vectors, which distinguish molecules also based on the amount of present
bits.</p>
<p>For count vectors the similarity metric is calculated using dice similarity instead of Tanimoto coefficient.</p>
<p>If multiple scaffolds are submitted a joint FP is computed for all scaffolds.
This allows SeMPI to match correct scaffolds also of clusters where only fragments could be predicted.</p>
<p>The FP screening only creates a pre-selection of possible scaffold matches (default 50).</p>
</div>
<div class="section" id="maximum-common-substructure-ranking">
<span id="mcs-rank"></span><h2>Maximum Common Substructure Ranking<a class="headerlink" href="#maximum-common-substructure-ranking" title="Permalink to this headline">¶</a></h2>
<p>Even though the FP screening collects very close matches to the predicted scaffolds,
some properties cannot be matched accurately with FPs.
For example the order of building blocks cannot be represented correctly even by count vectors.
Similar building blocks in different order can lead to wrong ranking solely based on fingerprints (see <a class="reference internal" href="#fp-example"><span class="std std-numref">Fig. 10</span></a>). Therefore an MCS based ranking is applied to the pre-selected molecules.</p>
<div class="figure align-default" id="id6">
<span id="fp-example"></span><a class="reference internal image-reference" href="../_images/fp_based_search_example.png"><img alt="../_images/fp_based_search_example.png" src="../_images/fp_based_search_example.png" style="width: 748.0px; height: 384.5px;" /></a>
<p class="caption"><span class="caption-number">Fig. 10 </span><span class="caption-text">Hypothetical screening of two building blocks of two alanine (block 0) and one serine (block 1) against
an two peptides (A-A-S) and (A-S-A). The peptide with the correct order of building blocks gets a lower similarity score (based on FPs), whereas the ConfS scores (based on MCS) represents the expected ranking.</span><a class="headerlink" href="#id6" title="Permalink to this image">¶</a></p>
</div>
<p>The maximum common substructure (MCS) algorithm ranks the pre-selection based on the
best MCS score for each scaffold fragment with the target
molecules.</p>
<div class="figure align-default" id="id7">
<span id="mcs-algo"></span><a class="reference internal image-reference" href="../_images/MCS_algo.svg"><img alt="../_images/MCS_algo.svg" src="../_images/MCS_algo.svg" /></a>
<p class="caption"><span class="caption-number">Fig. 11 </span><span class="caption-text">Example demonstration of the MCS algorithm. To simplify the example only two building blocks are used, the algorithm can potentially scale up to 10 building blocks. (1) Initially the building blocks are ordered by their number of atoms. The matches of the biggest building blocks are most meaning-full. (2) All MCS of the first block (B1) with the target molecule (Mol) are computed. The example shows only one MCS, but some molecules (especially ring systems) can have large amounts of MCS. The number of MCS to find for each B1 in a Mol are limited to 20. A new molecule is created for each MCS, where the MCS is removed from the scaffold. This new molecule is then submitted to a new MCS search with the next building block (B2).</span><a class="headerlink" href="#id7" title="Permalink to this image">¶</a></p>
</div>
</div>
</div>


            <div class="clearer"></div>
          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h3><a href="../index.html">Table of Contents</a></h3>
<p class="caption"><span class="caption-text">Description of the SeMPI 2.0 web-server</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="primer.html">Primer</a></li>
<li class="toctree-l1"><a class="reference internal" href="genome_input.html">Genome input</a></li>
<li class="toctree-l1"><a class="reference internal" href="prediction_algo.html">Prediction</a></li>
<li class="toctree-l1"><a class="reference internal" href="genome_browser.html">Genome browser</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Database screening</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#use-cases">Use cases</a></li>
<li class="toctree-l2"><a class="reference internal" href="#algorithm">Algorithm</a></li>
<li class="toctree-l2"><a class="reference internal" href="#scaffold-modification">Scaffold modification</a></li>
<li class="toctree-l2"><a class="reference internal" href="#custom-db-input-format">Custom DB input format</a></li>
<li class="toctree-l2"><a class="reference internal" href="#fingerprint-screening">Fingerprint screening</a></li>
<li class="toctree-l2"><a class="reference internal" href="#maximum-common-substructure-ranking">Maximum Common Substructure Ranking</a></li>
</ul>
</li>
</ul>
<p class="caption"><span class="caption-text">Update log</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="update_log.html">Updates</a></li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="genome_browser.html"
                        title="previous chapter">Genome browser</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="update_log.html"
                        title="next chapter">Updates</a></p>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="../search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" />
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="update_log.html" title="Updates"
             >next</a> |</li>
        <li class="right" >
          <a href="genome_browser.html" title="Genome browser"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="../index.html">SeMPI 2.0 documentation</a> &#187;</li>
        <li class="nav-item nav-item-this"><a href="">Database screening</a></li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &#169; Copyright 2019, Paul Zierep.
      Created using <a href="https://www.sphinx-doc.org/">Sphinx</a> 3.1.2.
    </div>
  </body>
</html>